### PR TITLE
Reactor: remove thread pool for runner/wheel

### PR DESCRIPTION
**Pull request by mattp-**
### What does this PR do?
rather than execute inline in a separate pool, we can push these off to the master to handle like other work, so lets do that instead. This deprecates/removes the REACTOR_WORKER_HWM and REACTOR_WORKER_THREADS settings.

also included:
add root_key identity check for interacting with the reactor
since actions through the reactor/manage interface can be potentially  destructive, we use the root_key as a identifier to assert it originated  only from the runner as the root user as opposed to any event coming through the bus.

### What issues does this PR fix or reference?
- performance bottleneck in the reactor engine.
- unauthenticated interaction with the reactor management interface

### Tests written?
Yes

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/utils/reactor.py
Comment (DmitryKuzmenko) [inline salt/utils/reactor.py]: Why we're getting to ignore the result here?

**Review by DmitryKuzmenko — COMMENTED** [review]
I agree at all, just please see one question inline.

File: salt/utils/reactor.py
Comment (mattp-) [reply salt/utils/reactor.py]: I guess the question is what should we do with the ret in reactors case? it was previously used to check it the Queue.Queue was full, but otherwise tossed. can we meaningfully do anything that isn't already emitted in the master log?

**Comment by mattp-** [comment]
@DmitryKuzmenko btw had some out of band chat with mike,
```Matthew Phillips [10:50 AM]
@cachedout re; our talk yesterday about the threadpool, i was mistaken it does actually use runnerclient.asynchronous (the fork/daemonize) when it processes runner jobs
i still dont like this though. what id like to do is move the queue idea out of the reactor and into the master itself
that way the zeromq channels can drain to this queue (could even be something like a priority queue, do this job before any other jobs if they come in etc)
it would be very easy to provide metrics on whats going on with jobs coming in if we do that, ie what is the queue depth, what is the contents etc
making the queue size 0 / infinite would leave the behavior backwards compatible, while also stopping runner/wheel jobs from forkbombing the server
the stats in particular have been something we’ve been looking for for quite a while, it just now clued to me how we can implement it. would like to hear your thoughts :+1:
Mike Place [11:07 AM]
I’m a fan of the queuing approach. Moving to that model and providing stats would be a huge win all around.
Matthew Phillips [1:16 PM]
indeed. ok ill try to hack on that at some point. in the interim please slap a hold on that PR
Mike Place [1:23 PM]
10-4```

**Comment by DmitryKuzmenko** [comment]
@mattp- the forks used everywhere is a big pain though.
What Mike said is a kind of what I'm trying to do in Tornado 5.0 work. So I agree.

**Review by DmitryKuzmenko — APPROVED** [review]

**Comment by cachedout** [comment]
@mattp- Hrm. We did see some issues with this: https://jenkinsci.saltstack.com/job/pr-kitchen-ubuntu1604-py3/job/PR-50675/13/testReport/junit/integration.reactor.test_reactor/ReactorTest/test_reactor_wheel_reaction/

**Comment by dwoz** [comment]
@mattp- Can you look into the conflicts?

**Comment by mattp-** [comment]
@dwoz please consider this on hold for time being. I need to sort out a similar throttle on master itself before this is safe for merge

**Comment by dwoz** [comment]
@mattp- Can we close this for now and once you get things resolve make another PR?

**Comment by mattp-** [comment]
yep thats fine, closing

**Pull request closed without merging**
